### PR TITLE
[JUJU-1921] Add GitHub smoke tests for MicroK8s

### PR DESCRIPTION
**Pull request by barrettj12**
Title.

Drive-by smoke test improvements:
- Fix typo in `workload_status`
- Call `juju remove-application` before `destroy_model`. This fixes an issue where the model teardown times out on MicroK8s because the application is still live.

## QA steps

Wait for "Smoke / Smoke (microk8s)" workflow to pass below.

### CODE REVIEW

File: .github/workflows/smoke.yml [hunk 1]
```diff
@@ -35,8 +39,16 @@ jobs:
...
 
     - name: Setup LXD
+      if: matrix.cloud == 'localhost'
       uses: canonical/setup-lxd@90d76101915da56a42a562ba766b1a77019242fd
 
+    - name: Setup MicroK8s
+      if: matrix.cloud == 'microk8s'
+      uses: balchua/microk8s-actions@v0.3.1
+      with:
+        channel: "1.23/stable"
```
Comment (ycliuhw) [inline .github/workflows/smoke.yml:49]: We can install `1.25/stable`, and enable `dns`, `hostpath-storage`(storage is deprecated), and `RBAC`(then we can test RBAC).

File: .github/workflows/smoke.yml [hunk 2]
```diff
@@ -53,8 +65,21 @@ jobs:
       run: |
         make go-install
 
-    - name: Smoke Test
+    - name: Update microk8s operator image
+      if: matrix.cloud == 'microk8s'
+      run: |
+        make microk8s-operator-update
```
Comment (ycliuhw) [inline .github/workflows/smoke.yml:71]: Would be better to use a fake docker account to ensure we are testing the locally built image.

File: .github/workflows/smoke.yml [hunk 3]
```diff
@@ -53,8 +65,21 @@ jobs:
...
       run: |
         cd tests
         ./main.sh -v -s 'test_build' smoke
+
+    - name: Smoke test (MicroK8s)
+      if: matrix.cloud == 'microk8s'
+      shell: bash
+      run: |
+        cd tests
+        sg microk8s './main.sh -c microk8s -s test_build -v smoke'
```
Comment (ycliuhw) [inline .github/workflows/smoke.yml:85]: Inject `--config caas-image-repo foo`

File: tests/suites/smoke/deploy.sh [hunk 4]
```diff
@@ -16,8 +16,9 @@ run_local_deploy() {
 	juju refresh refresher
 
 	# Wait for the refresh to happen and then wait again.
-	wait_for "upgrade hook ran v2" "$(workloadstatus "refresher" 0)"
+	wait_for "upgrade hook ran v2" "$(workload_status "refresher" 0)"
```
Comment (ycliuhw) [inline tests/suites/smoke/deploy.sh:19]: Good catch.

File: tests/suites/smoke/deploy.sh [hunk 5]
```diff
@@ -16,8 +16,9 @@ run_local_deploy() {
...
 
+	juju remove-application refresher
```
Comment (ycliuhw) [inline tests/suites/smoke/deploy.sh:21]: Probably leave a TODO for removing in the future.

**Review by ycliuhw — APPROVED** [review]

File: tests/suites/smoke/deploy.sh (hunk 5 above)
Comment (barrettj12) [reply tests/suites/smoke/deploy.sh:21]: I don't see the issue. It's probably good practice to remove the application before tearing down the model anyway

Comment (ycliuhw) [reply tests/suites/smoke/deploy.sh:21]: This is definitely a bug, so removing app before teardown is a temp fix.
We should remove the temp fix once we fix the teardown issue.

**Pull request merged**